### PR TITLE
Prefer project npmrc if present

### PR DESCRIPTION
**Pull request by zephraph**
# What Changed

`.npmrc` files now prefer the project directory over the home directory. If there's no `.npmrc` in the project directory, but there _is_ in the home directory, it'll fall back to use that one.

## Why

Often when someone writes a `.npmrc` from CI, they just write it in place. That's valid to npm, but not recognized by auto. If a user writes a local .npmrc file, that'll be preferred over the global one (in the home directory). This could lead to situations where the `.npmrc` file gets written by auto, but _doesn't_ get used by the project. 

Todo:

- [x] Add tests
- [x] Add docs

## Change Type

Indicate the type of change your pull request is:

- [ ] `documentation`
- [ ] `patch`
- [x] `minor`
- [ ] `major`

### CODE REVIEW

**Comment by zephraph** [comment]
Welp, found my first bug?

```
?? packages/cli/.npmrc
Error: Working directory is not clean, make sure all files are committed
```

Looks like `process.cwd()` is _not_ the right way to go here...

File: plugins/npm/src/set-npm-token.ts [hunk 1]
```diff
@@ -45,35 +50,46 @@ export default async function setTokenOnCI(logger: ILogger) {
...
 
-  try {
-    contents = (await readFile(rc)).toString();
-  } catch (error) {
-    // No ~/.npmrc set up
+  const [readRCError, npmrcContents = ""] = await to(
+    readFile(npmrcFilePath, "utf-8")
+  );
+  if (readRCError) {
+    logger.verbose.info(`${npmrcFilePath} doesn't exist or cannot be read`);
```
Comment (hipstersmoothie) [inline plugins/npm/src/set-npm-token.ts:59]: could add the error using logger.veryVerbose.error(error)

File: plugins/npm/src/utils.ts [hunk 2]
```diff
@@ -1,17 +1,70 @@
...
+  const result = pathToLerna ? path.join(pathToLerna, "lerna.json") : false;
+  if (process.env.NODE_ENV !== "test") {
+    getLernaPathIfExists._path = result;
+  }
+  return result;
+};
+
 /** Check if the project is a monorepo */
-export const isMonorepo = () => fs.existsSync("lerna.json");
+export const isMonorepo = (): boolean => !!getLernaPathIfExists();
```
Comment (hipstersmoothie) [inline plugins/npm/src/utils.ts:43]:
```suggestion
export const isMonorepo = (): boolean => Boolean(getLernaPathIfExists());
```

**Review by hipstersmoothie — CHANGES_REQUESTED** [review]
Looks good, just a few changes

**Review by hipstersmoothie — APPROVED** [review]

**Comment by hipstersmoothie** [comment]
@zephraph could you fix the failing test?

**Comment by zephraph** [comment]
🤞

**Comment by zephraph** [comment]
I haven't touched this in a while. I might revisit one day, but I'm going to close this for today.

**Pull request closed without merging**